### ISSUE TITLE
Updating a KVM does not work

### ISSUE THREAD

**Issue by arnaduga** [scastria/terraform-provider-apigee#2]
When I try to update a KVM values, I got this error:
```shell
Error: Status 410: Message: Gone: {
  "code" : "keyvaluemap.service.KeyValueMapUpdateOperationNotSupported",
  "message" : "Update of KeyValueMap is not supported; use addEntry/updateEntry api to add entries to a KeyValueMap",
  "contexts" : [ ]
}
```

Code used:
```ruby
terraform {
  required_providers {
    apigee = {
      source  = "scastria/apigee"
      version = "~> 0.1.0"
    }
  }
}

# Configure the Apigee Provider
provider "apigee" {
  version  = "~> 0.1.0"
  username = "email@example.cm"
  password = "J#EgED&!c^AyqM6U*SFgP9sEefs3CADg"
  server       = "api.enterprise.apigee.com"
  organization = "myOrg"
}

resource "apigee_environment_kvm" "example" {
  environment_name = "test"
  name             = "LookupValues"
  entry = {
    first  = "firstValue"
    second = "secondValue"
  }
}
```

... then updated to:
```ruby
...
  entry = {
    first  = "firstValue_new"
    second = "secondValue_new"
  }
...
```

Versions:
```
 ~/$ terraform --version
Terraform v0.13.2
+ provider registry.terraform.io/scastria/apigee v0.1.16
```

**Comment by scastria** [comment]
Thanks for pointing that out.  It looks the supported API calls to the Apigee management server is different based on whether your Apigee installation is private or public.  The Private version (the one I developed this provider for) DOES support modifying entries in bulk, but the Public version (the one you are probably using) DOES NOT.  The public version requires a change to an entry to be done individually.  Therefore, my provider needs to know whether the Apigee server it is talking to is private or not so it can do the right thing.

Do you think I should just compare the server hostname to "api.enterprise.apigee.com" to know that it is public?  I worry about this as Google may change the hostname in the future.  Another option is to assume public but add another optional provider configuration property called is_private where the user of my provider must tell me it is private or not.  What do you think?

**Comment by arnaduga** [comment]
I never used an onPrem version ;)

I do not like the idea of having a `_private`... seems inappropriate to me  to split like that.

I do not think the `api.enterprise.apigee.com` URL would change, _however_, (unfortunately), I cannot predict future :-)

So, maybe this could be done (just a proposal):
- make the server parameter optional (and equal to "api.enterprise.apigee.com" when not populated as it _should not_ change)
- add another parameter `private` or `saas_mode`, that would be valued `false`/`true`respectively when "public" and `true`/`false`when "on prem"?

Then, if private, you can do the same as today, if not, play with the other endpoint.

**Does that make sense?**


***Question however***: in the private edition, does the APIGEE [_addEntry/updateEntry api_](https://apidocs.apigee.com/docs/key-value-maps/1/routes/organizations/%7Borg_name%7D/keyvaluemaps/%7Bmap_name%7D/entries/%7Bentry_name%7D/get) work? If yes, why do not apply this method, whatever it is public or private?  No distinction between both, easier...


In the meantime, the workaround I found is to _taint_ the resource:
`terraform taint apigee_environment_kvm.example`, 
and then the `terraform apply` will force the deletion and rebuild the entire KVM... Not perfect, but taht could help ;)

Thanks

**Comment by scastria** [comment]
I had just tried that. The individual entry api ONLY works for public and the bulk entry api ONLY works for private so there is no single way to make it work for both. I will default the server hostname as you suggest and add a “private” config parameter and then implement different logic based on public or private. Should be done today.

**Comment by arnaduga** [comment]
Oh... Not very convenient to have different behaviour like that...

Thank you very much for your reactivity!

**Comment by scastria** [comment]
Hopefully, v0.1.17 will do the trick when it is done building.

**Comment by scastria** [comment]
Actually, I think I changed my mind on the private flag.  That will force all of the OnPrem people, like me, to have to make changes to their terraform scripts and set private = true.  So think the most flexible solution is to keep the flag but make the default more advanced.  If the private flag is specified (in TF template or via APIGEE_PRIVATE env variable), then honor it.  If not specified, then compare the server name to api.enterprise.apigee.com to obtain default value.

**Comment by scastria** [comment]
It doesn't look like this is supported by the provider SDK.  So I think I will just remove the private parameter for now and assume api.enterprise.apigee.com won't change.  I can always add the private parameter later if needed.

**Comment by scastria** [comment]
v0.1.18 is building now without the `private` provider config pparameter

**Comment by arnaduga** [comment]
I tested the v0.1.18 with no server set, for add, changing and removing a key in the KVM and it worked perfectly!

Just a remark: doc on [Terraform Registry](https://registry.terraform.io/providers/scastria/apigee/latest/docs) is not aligned with you recent changes. The server and port and mentioned as **Required**.

I guess however the issue can be closed. Many thanks!

**Comment by scastria** [comment]
Fix server and port to be Optional for v0.1.19
